tests/are_images_equal.cc

```diff
@@ -0,0 +1,34 @@
+// Copyright 2022 Google LLC. All rights reserved.
```

Comment:
Our current policy allows C++ code only in the tests/gtest directory. Since this file doesn't use GoogleTest, it may seem inappropriate to put this file in tests/gtest.

One solution is to move the `libavif::testutil::AreImagesEqual()` function to apps/shared/avifutil.h. We will need to convert it to a C function, and therefore won't be able to specify a default argument for the `ignore_alpha` parameter.

Another solution is to move this file to tests/gtest.

Note: We can also write the lossless test using GoogleTest. Then we don't need to add this file.

Reply:
The test is about testing avifenc/avifdec, not the algorithms. Moving it to the gtest folder.